### PR TITLE
pythonPackage shortuuid: init at 0.4.2

### PR DESCRIPTION
**Pull request by zagy**

### CODE REVIEW

**Comment by domenkozar** [comment]
`hash mismatch in downloaded path ‘/nix/store/a45l8r0zqfr5jhqswqqcya28h59n63js-python2.7-setuptools-18.2’: expected 563b3d1b693c5579ef6f05544ff72fc30a26e13e506ee5265eea81081c331b6d, got 50db90fe916eca8cbba2f3a1a00dae35c4f3d40a25ff693487938bf38ae8ff3a`

This is the same as you got @FRidh right?

**Comment by FRidh** [comment]
Yes, exactly. https://github.com/NixOS/nixpkgs/issues/11320
It was odd, it happened I think twice after each other but after that I wasn't able to reproduce it.

**Comment by FRidh** [comment]
also the exact same hashes

**Comment by zagy** [comment]
Oh boy. Could the CDN or some cache have a broken version? That's what the checksum are for after all.

**Comment by zagy** [comment]
Anything I can  do here?

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -16728,6 +16728,27 @@ in modules // {
...
 
+  shortuuid = buildPythonPackage rec {
+    name = "shortuuid-${version}";
+    version = "0.4.2";
+
+    disabled = isPy26;
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/s/shortuuid/${name}.tar.gz";
+      md5 = "142e3ae4e7cd32d41a71deb359db4cfe";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you use a sha256 instead? It's the preferred hash.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -16728,6 +16728,27 @@ in modules // {
...
+    };
+
+    buildInputs = with self; [pep8];
+
+    meta = {
+      description = "A generator library for concise, unambiguous and URL-safe UUIDs.";
+      homepage = https://github.com/stochastic-technologies/shortuuid/;
+      license = licenses.bsd3;
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: perhaps you are interested in adding yourself as maintainer of this package?

**Comment by FRidh** [comment]
Looks fine to me. Did you test whether it builds for all interpreters and versions?

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (zagy) [inline pkgs/top-level/python-packages.nix]: Generally yes. The md5 is the one which is published on pypi, making it trivial to add. So if it's not a huge problem, I'd stick with that.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (zagy) [inline pkgs/top-level/python-packages.nix]: I thought I did. But I'm not sure anymore. Will re-check.

**Comment by zagy** [comment]
I thought I did. But I'm not sure anymore. Will re-check.

**Comment by zagy** [comment]
Okay, now I did something with git which was probably wrong. :/ 
Will check on that later.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -16728,6 +16728,48 @@ in modules // {
...
+  };
+
+
+  shortuuid = buildPythonPackage rec {
```
Comment (garbas) [inline pkgs/top-level/python-packages.nix]: why duplicating the package?

**Comment by zagy** [comment]
better now?

**Comment by garbas** [comment]
@zagy still no :( check travis build or use nox locally to verify...

currently it looks like you're using isPy32 which was removed

**Comment by zagy** [comment]
> On 07.12.2015, at 14:25, Rok Garbas <notifications@github.com> wrote:
> 
> @zagy still no :( check travis build or use nox locally to verify...
> 
> currently it looks like you're using isPy32 which was removed

Yeah, sorry. My workflow and knowledge aren't good enough, yet. Didn’t even know about nox. Will try that. Thanks for your patience.





-- 
Dipl. Inform. (FH) Christian Zagrodnick · cz@flyingcircus.io · +49 345 219401 0
Flying Circus Internet Operations GmbH · http://flyingcircus.io
Forsterstraße 29 · 06112 Halle (Saale) · Deutschland
HR Stendal HRB 21169 · Geschäftsführer: Christian Theune, Christian Zagrodnick

**Comment by vcunat** [comment]
Ping.

**Comment by zagy** [comment]
@vcunat Hi. Sorry for the late response. There was this state where master wasn't building (the bash thing) and I just couldn't get to the point of testing this reliably.

**Comment by domenkozar** [comment]
@zagy I've merged this, but please use `sha256` as we're trying to phase out md5 https://github.com/NixOS/nixpkgs/issues/4491

**Pull request merged**